src/proxy.rs

```diff
@@ -114,31 +118,28 @@ fn eat_type(iter: &mut Peekable<token_stream::IntoIter>) -> Result<ProxyType, ()
             }
         }
 
-        TokenTree::Ident(ident) => {
-            match &*ident.to_string() {
-                "Box" => ProxyType::Box,
-                "Rc" => ProxyType::Rc,
-                "Arc" => ProxyType::Arc,
-                "Fn" => ProxyType::Fn,
-                "FnMut" => ProxyType::FnMut,
-                "FnOnce" => ProxyType::FnOnce,
-                _ => {
-                    let msg = format!("unexpected '{}'. {}", ident, EXPECTED_TEXT);
-                    ident.span()
-                        .err(msg)
-                        .note(NOTE_TEXT)
-                        .emit();
-
-                    return Err(());
-                }
+        TokenTree::Ident(ident) => match &*ident.to_string() {
+            "Box" => ProxyType::Box,
+            "Rc" => ProxyType::Rc,
+            "Arc" => ProxyType::Arc,
+            "Fn" => ProxyType::Fn,
+            "FnMut" => ProxyType::FnMut,
+            "FnOnce" => ProxyType::FnOnce,
+            _ => {
+                emit_error!(
+                    ident.span(),
+                    "unexpected '{}', {}", ident, EXPECTED_TEXT;
+                    note = NOTE_TEXT
+                );
+
+                return Err(());
```

Comment:
Please use the old extra-braces syntax: `match _ { Ident(ident) => { match ...`. I doubt this change was made by rustfmt?

Reply:
`rustfmt` is the one to blame here. I would argue that the current approach is somewhat cleaner though, what do you think?